### PR TITLE
use op-alloy genesis types for genesis parsing

### PR DESCRIPTION
**Pull request by nkysg**
Fix the issue at https://github.com/paradigmxyz/reth/issues/9267

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -419,6 +419,9 @@ alloy-pubsub = { version = "0.1", default-features = false }
 alloy-json-rpc = { version = "0.1", default-features = false }
 alloy-rpc-client = { version = "0.1", default-features = false }
 
+#op
+op-alloy-rpc-types = { version = "0.1" }
```
Comment (mattsse) [inline Cargo.toml:423]:
```suggestion
# op
op-alloy-rpc-types = "0.1"
```

File: crates/chainspec/src/spec.rs [hunk 2]
```diff
@@ -1085,32 +1088,21 @@ impl DepositContract {
 #[derive(Default, Debug, serde::Deserialize)]
 #[serde(rename_all = "camelCase")]
 struct OptimismGenesisInfo {
-    bedrock_block: Option<u64>,
-    regolith_time: Option<u64>,
-    canyon_time: Option<u64>,
-    ecotone_time: Option<u64>,
-    fjord_time: Option<u64>,
+    genesis_info: OpGenesisInfo,
```
Comment (mattsse) [inline crates/chainspec/src/spec.rs:1091]: we can use this: OptimismChainInfo

File: crates/chainspec/src/spec.rs [hunk 3]
```diff
@@ -1085,32 +1088,21 @@ impl DepositContract {
...
-            genesis.config.extra_fields.deserialize_as().unwrap_or_default();
+        let mut optimism_genesis_info = OptimismGenesisInfo::default();
+        if let Some(genesis_info) = OpGenesisInfo::extract_from(&genesis.config.extra_fields) {
+            optimism_genesis_info.genesis_info = genesis_info;
+        }
 
-        if let Some(Ok(optimism_base_fee_info)) =
-            genesis.config.extra_fields.get_deserialized::<OptimismBaseFeeInfo>("optimism")
+        if let Some(optimism_base_fee_info) =
+            OptimismBaseFeeInfo::extract_from(&genesis.config.extra_fields)
```
Comment (mattsse) [inline crates/chainspec/src/spec.rs:1105]: this can be collapsed into `OptimismChainInfo::extract_from`

File: crates/chainspec/Cargo.toml [hunk 4]
```diff
@@ -24,6 +24,9 @@ alloy-genesis.workspace = true
 alloy-primitives = { workspace = true, features = ["rand", "rlp"] }
 alloy-trie.workspace = true
 
+# op
+op-alloy-rpc-types.workspace = true
```
Comment (mattsse) [inline crates/chainspec/Cargo.toml:28]: this should be optional and only enabled with the op feature

**Review by mattsse — CHANGES_REQUESTED** [review]
awesome, some suggestions

File: Cargo.toml (hunk 1 above)
Comment (nkysg) [reply Cargo.toml:423]: fixed

File: crates/chainspec/src/spec.rs (hunk 2 above)
Comment (nkysg) [reply crates/chainspec/src/spec.rs:1091]: ok. thanks

File: crates/chainspec/src/spec.rs (hunk 3 above)
Comment (nkysg) [reply crates/chainspec/src/spec.rs:1105]: fixed. thx

File: crates/chainspec/src/spec.rs [hunk 5]
```diff
@@ -39,6 +39,9 @@ use reth_network_peers::{
     op_testnet_nodes, sepolia_nodes,
 };
 
+#[cfg(feature = "optimism")]
+use op_alloy_rpc_types::genesis::OptimismChainInfo;
```
Comment (mattsse) [inline crates/chainspec/src/spec.rs:43]: please use full path instead of feature gating an import

File: Cargo.toml [hunk 6]
```diff
@@ -419,6 +419,9 @@ alloy-pubsub = { version = "0.1", default-features = false }
 alloy-json-rpc = { version = "0.1", default-features = false }
 alloy-rpc-client = { version = "0.1", default-features = false }
 
+# op
+op-alloy-rpc-types = { version = "0.1", features = ["arbitrary"] }
```
Comment (mattsse) [inline Cargo.toml:423]:
```suggestion
op-alloy-rpc-types = "0.1"
```

File: crates/chainspec/Cargo.toml [hunk 7]
```diff
@@ -40,11 +44,15 @@ alloy-genesis.workspace = true
...
+
 [features]
 default = ["std"]
 optimism = [
     "reth-ethereum-forks/optimism",
-    "serde"
+    "serde",
+    "op-alloy-rpc-types/arbitrary",
```
Comment (mattsse) [inline crates/chainspec/Cargo.toml:55]: needs "dep:op-alloy-rpc-types"

**Review by mattsse — CHANGES_REQUESTED** [review]
cool, last pedantic dep nits

File: crates/chainspec/Cargo.toml (hunk 7 above)
Comment (nkysg) [reply crates/chainspec/Cargo.toml:55]: Fixed. Thx

File: Cargo.toml (hunk 6 above)
Comment (nkysg) [reply Cargo.toml:423]: Fixed. Thx!

File: crates/chainspec/src/spec.rs (hunk 5 above)
Comment (nkysg) [reply crates/chainspec/src/spec.rs:43]: Fixed. Thx

**Review by mattsse — APPROVED** [review]
ty

**Pull request merged**